client/app/bundles/course/experience-points/components/ExperiencePointsTableRow.tsx

```diff
@@ -0,0 +1,196 @@
+import { FC, memo, useEffect, useState } from 'react';
+import { TableCell, TableRow } from '@mui/material';
+import equal from 'fast-deep-equal';
+import {
+  ExperiencePointsRecordMiniEntity,
+  ExperiencePointsRowData,
+} from 'types/course/experiencePointsRecords';
+
+import PointManagementButtons from 'course/users/components/buttons/PointManagementButtons';
+import NumberTextField from 'lib/components/core/fields/NumberTextField';
+import TextField from 'lib/components/core/fields/TextField';
+import Link from 'lib/components/core/Link';
+import { formatMiniDateTime } from 'lib/moment';
+
+interface Props {
+  id: number;
+  isStudentPage?: boolean;
+  record: ExperiencePointsRecordMiniEntity;
+  maxExp?: number;
+  isDownloading?: boolean;
+}
+
+const ExperiencePointsTableRow: FC<Props> = (props) => {
+  const { record, id, isStudentPage, maxExp, isDownloading } = props;
+  const [isDirty, setIsDirty] = useState(false);
+  const [errorHelperText, setErrorHelperText] = useState('');
+  const [errorReasonText, setErrorReasonText] = useState('');
+  const [rowData, setRowData] = useState({
+    id,
+    reason: '',
+    pointsAwarded: 0,
+  } as ExperiencePointsRowData);
+  const [defaultRowData, setDefaultRowData] = useState({
+    id,
+    reason: '',
+    pointsAwarded: 0,
+  } as ExperiencePointsRowData);
+  const isNotManuallyAwarded = maxExp;
```

Comment:
Why is this called `isNotManuallyAwarded` but it takes the value of `maxExp`?

Reply:
the intention is basically to store the boolean value in which maxExp is defined or not (if it has value, then that isNotManuallyAwarded is true, otherwise false)..

for clarity, I changed the logic a bit to emphasise on that

Reply:
But there’s ismanuallyawarded in pointsreason?